readthedocs/projects/forms.py

```diff
@@ -614,12 +614,14 @@ def get_translation_queryset(self):
         )
         return queryset
 
-    def save(self):
-        project = self.parent.translations.add(self.translation)
-        # Run symlinking and other sync logic to make sure we are in a good
-        # state.
-        self.parent.save()
-        return project
+    def save(self, commit=True):
+        if commit:
+            self.translation.save()
```

Comment:
Would you like to add a comment explaining a little more why this is required here?

Reply:
I made the fix clearer. The problem is that when calling `add()`, `self.parent` isn't the object that is updated. The field that is updated is `self.translation.main_language_project = self.parent`. Calling `save()` first was a method to make the object state use the database that matches `db_for_write()`. Reversing this logic makes this less of a hack and skips usage of `add()` altogether.

It's not entirely clear why this is an issue on only the translation relationship however.